extensions/amp-base-carousel/0.1/test-e2e/test-default-attributes.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describes.endtoend(
+  'AMP carousel test default attributes',
+  {
+    testUrl:
+      'http://localhost:8000/test/manual/amp-base-carousel/default-attributes.amp.html',
+    environments: ['single'],
+  },
+  async function (env) {
+    let controller;
+    let scrollContainer;
+    let loop;
+
+    beforeEach(async () => {
+      controller = env.controller;
+    });
+
+    function getScrollContainerByCarouselId(id) {
+      return controller.findElement(
+        `amp-base-carousel#${id} .i-amphtml-carousel-scroll`
+      );
+    }
+
+    it('should add loop="false" when loop is omitted (carousel-1)', async () => {
+      scrollContainer = await getScrollContainerByCarouselId('carousel-1');
+      loop = await controller.getElementAttribute(scrollContainer, 'loop');
+      await expect(loop).to.be.equal('false');
+    });
+
+    it('should keep loop="true" when valid (carousel-2)', async () => {
+      scrollContainer = await getScrollContainerByCarouselId('carousel-2');
+      loop = await controller.getElementAttribute(scrollContainer, 'loop');
+      await expect(loop).to.equal('true');
+    });
+
+    it('should set loop to "false" when invalid (carousel-3)', async () => {
+      scrollContainer = await getScrollContainerByCarouselId('carousel-3');
+      loop = await controller.getElementAttribute(scrollContainer, 'loop');
+      await expect(loop).to.equal('false');
+    });
+  }
```

Comment:
What do you think about adding a media-query based `loop` attr case? I saw a similar early comment was marked unneeded, could you elaborate why? 

For example, we could set a default document size (800px by 800px), and `loop="(min-width: 1000px) true, false"`. Then assert that `loop="false"` on the scroller by default, resize to >1000px, then assert that it has been correctly updated.

Reply:
Ooops sorry about that. 

Responsive attributes cause the carousel to update `this.loop_` and then trigger `carousel#updateUi()` which then set the `loop` attribute.

User set `loop` attribute will always be overrode if invalid. 

> What do you think about adding a media-query based loop attr case? 

This is def a valid case! I'll add in another test for the extra insurance :D